graylog2-web-interface/src/hooks/useHotkeys.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import type { HotkeyCallback, Options } from 'react-hotkeys-hook';
+import { useHotkeys as originalUseHotkeys } from 'react-hotkeys-hook';
+import { useEffect, useMemo } from 'react';
+
+import type { ScopeName } from 'contexts/HotkeysContext';
+import useHotkeysContext from 'hooks/useHotkeysContext';
+
+const defaultOptions: Options = {
+  preventDefault: true,
+  enabled: true,
+  enableOnFormTags: false,
+  enableOnContentEditable: false,
+  combinationKey: '+',
+  splitKey: ',',
+  scopes: '*',
+  keyup: undefined,
+  keydown: true,
+  description: undefined,
+  document: undefined,
+  ignoreModifiers: false,
+};
+
+const useHotkeys = <T extends HTMLElement>(
+  actionKey: string,
+  callback: HotkeyCallback,
+  options?: Options & { scopes: ScopeName },
+  dependencies?: Array<unknown>,
+) => {
```

Comment:
Because we talked about the parameters of this function. I like the following approach:
```
scope: string,
actionKey: string,
callback: HotkeyCallback,
options?: { dependencies: ScopeName }
```
Se the usage would look like this:
`useHotkeys('dashboard', 'SAVE', _onSaveView);` instead of
`useHotkeys('SAVE', _onSaveView, { scopes: 'dashboard' });`

Reply:
Now I saw that `scopes` is part of the `react-hotkeys-hook` options argument.